examples/manipulation/grasp_env.py

```diff
@@ -0,0 +1,510 @@
+            show_viewer=show_viewer,
+        )
+
+        # == add ground ==
+        self.scene.add_entity(gs.morphs.URDF(file="urdf/plane/plane.urdf", fixed=True))
+
+        # == add robot ==
+        self.robot = Manipulator(
+            num_envs=self.num_envs,
+            scene=self.scene,
+            args=robot_cfg,
+            device=gs.device,
+        )
+
+        # == add object ==
+        self.object = self.scene.add_entity(
+            gs.morphs.Box(
+                size=env_cfg["box_size"],
+                fixed=env_cfg["box_fixed"],
+                collision=env_cfg["box_collision"],
+            ),
+            # material=gs.materials.Rigid(gravity_compensation=1),
+            surface=gs.surfaces.Rough(
+                diffuse_texture=gs.textures.ColorTexture(
+                    color=(1.0, 0.0, 0.0),
+                ),
+            ),
+        )
+
+        # == add camera ==
+        if self.env_cfg["visualize_camera"]:
+            self.cam = self.scene.add_camera(
+                res=(1280, 720),
+                pos=(1.5, 0.0, 0.2),
+                lookat=(0, 0, 0.2),
+                fov=50,
+                GUI=True,
+            )
+
+        # build
+        self.scene.build(n_envs=num_envs)
+        # set pd gains (must be called after scene.build)
+        self.robot.set_pd_gains()
+
+        # prepare reward functions and multiply reward scales by dt
+        self.reward_functions, self.episode_sums = dict(), dict()
+        for name in self.reward_scales.keys():
+            self.reward_scales[name] *= self.ctrl_dt
+            self.reward_functions[name] = getattr(self, "_reward_" + name)
+            self.episode_sums[name] = torch.zeros((self.num_envs,), device=gs.device, dtype=gs.tc_float)
+
+        self.keypoints_offset = self.get_keypoint_offsets(batch_size=self.num_envs, device=self.device, unit_length=0.5)
+        # == init buffers ==
+        self._init_buffers()
+
+    def _init_buffers(self):
+        self.episode_length_buf = torch.zeros((self.num_envs,), device=gs.device, dtype=gs.tc_int)
+        self.reset_buf = torch.zeros(self.num_envs, dtype=torch.bool, device=gs.device)
+        self.goal_pose = torch.zeros(self.num_envs, 7, device=gs.device)
+        #
+        self.extras = dict()
+        self.extras["observations"] = dict()
+
+    def reset_idx(self, envs_idx):
+        if len(envs_idx) == 0:
+            return
+        self.episode_length_buf[envs_idx] = 0
+
+        # reset robot
+        self.robot.reset(envs_idx)
+
+        # reset object
+        num_reset = len(envs_idx)
+        random_x = torch.rand(num_reset, device=self.device) * 0.4 + 0.2  # 0.2 ~ 0.6
+        random_y = (torch.rand(num_reset, device=self.device) - 0.5) * 0.5  # -0.25 ~ 0.25
+        random_z = torch.ones(num_reset, device=self.device) * 0.025  # 0.15 ~ 0.15
+        random_pos = torch.stack([random_x, random_y, random_z], dim=-1)
+
+        #
```

Comment:
can you add comments or delete it.

Reply:
done. deleted all redundant comments symbols without comments